electronics_lib/test_JlcCapacitor.py

```diff
@@ -0,0 +1,70 @@
+
+  def test_capacitor_part(self) -> None:
+    compiled = ScalaCompiler.compile(JlcCapacitorTestTop, Refinements(
+        instance_values=[(['dut', 'part_spec'], '0603B103K500NT')]
+    ))
+    self.assertEqual(compiled.get_value(['dut', 'footprint_name']), 'Capacitor_SMD:C_0603_1608Metric')
+    self.assertEqual(compiled.get_value(['dut', 'part']), '0603B103K500NT')
+    self.assertEqual(compiled.get_value(['dut', 'value']), 'X7R ±10% 50V 10nF 0603 Multilayer Ceramic Capacitors MLCC - SMD/SMT ROHS')
+
+  def test_capacitor_footprint(self) -> None:
+    compiled = ScalaCompiler.compile(JlcCapacitorTestTop, Refinements(
+        instance_values=[(['dut', 'footprint_spec'], 'Capacitor_SMD:C_0603_1608Metric')]
+    ))
+    self.assertEqual(compiled.get_value(['dut', 'footprint_name']), 'Capacitor_SMD:C_0603_1608Metric')
+    self.assertEqual(compiled.get_value(['dut', 'part']), '0603B103K500NT')
+    self.assertEqual(compiled.get_value(['dut', 'value']), 'X7R ±10% 50V 10nF 0603 Multilayer Ceramic Capacitors MLCC - SMD/SMT ROHS')
+
+  def test_multi_capacitor(self) -> None:
+    compiled = ScalaCompiler.compile(JlcBigCapacitorTestTop)
+    self.assertEqual(compiled.get_value(['dut', 'c[0]', 'footprint_name']), 'Capacitor_SMD:C_0805_2012Metric')
+    self.assertEqual(compiled.get_value(['dut', 'c[0]', 'part']), 'CL21A106KAYNNNE')
+    self.assertEqual(compiled.get_value(['dut', 'c[0]', 'value']), 'X5R 25V ±10% 10uF 0805 Multilayer Ceramic Capacitors MLCC - SMD/SMT ROHS')
+    self.assertEqual(compiled.get_value(['dut', 'c[1]', 'footprint_name']), 'Capacitor_SMD:C_0805_2012Metric')
+    self.assertEqual(compiled.get_value(['dut', 'c[1]', 'part']), 'CL21A106KAYNNNE')
+    self.assertEqual(compiled.get_value(['dut', 'c[1]', 'value']), 'X5R 25V ±10% 10uF 0805 Multilayer Ceramic Capacitors MLCC - SMD/SMT ROHS')
+    self.assertEqual(compiled.get_value(['dut', 'c[2]', 'footprint_name']), 'Capacitor_SMD:C_0805_2012Metric')
+    self.assertEqual(compiled.get_value(['dut', 'c[2]', 'part']), 'CL21A106KAYNNNE')
+    self.assertEqual(compiled.get_value(['dut', 'c[2]', 'value']), 'X5R 25V ±10% 10uF 0805 Multilayer Ceramic Capacitors MLCC - SMD/SMT ROHS')
+    self.assertEqual(compiled.get_value(['dut', 'c[3]', 'footprint_name']), 'Capacitor_SMD:C_0805_2012Metric')
+    self.assertEqual(compiled.get_value(['dut', 'c[3]', 'part']), 'CL21A106KAYNNNE')
+    self.assertEqual(compiled.get_value(['dut', 'c[3]', 'value']), 'X5R 25V ±10% 10uF 0805 Multilayer Ceramic Capacitors MLCC - SMD/SMT ROHS')
+    self.assertEqual(compiled.get_value(['dut', 'c[4]', 'footprint_name']), 'Capacitor_SMD:C_0805_2012Metric')
+    self.assertEqual(compiled.get_value(['dut', 'c[4]', 'part']), 'CL21A106KAYNNNE')
+    self.assertEqual(compiled.get_value(['dut', 'c[4]', 'value']), 'X5R 25V ±10% 10uF 0805 Multilayer Ceramic Capacitors MLCC - SMD/SMT ROHS')
+    self.assertEqual(compiled.get_value(['dut', 'c[5]', 'footprint_name']), 'Capacitor_SMD:C_0805_2012Metric')
```

Comment:
I was hoping you could explain it to me. I expected 5 capacitors at first, but there was a signal that sixth one exists.

Reply:
So 25V is just a voltage rating - the capacitor is happy with voltages up to 25V, so it'll be perfectly fine if we only apply 3.3V.

It might be worth debugging why 6 capacitors are generated. Is it derating the capacitance such that it thinks we need 6? Or is something else going on? Might be worth dumping the part table row to see what the parallel column values turned out to be?